bip-newproposal.mediawiki

```diff
@@ -0,0 +1,140 @@
+  Status: Draft
+  Type: Standards Track
+  Created: 2025-07-02
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This BIP proposes '''PQ-Derived Schnorr''', a hybrid signature scheme that cryptographically binds a post-quantum signature (CRYSTALS-Dilithium) with Schnorr signatures (BIP340).  
+The scheme enables post-quantum secure transactions while maintaining compatibility with existing Bitcoin infrastructure.  
+
+Verification on-chain uses only a 64-byte Schnorr segment, while full Dilithium verification can be enforced by Tier-2/archive nodes.  
+The proposal introduces a new Taproot leaf version and an optional <code>OP_CHECKSIG_QES2</code> opcode to support hybrid verification.
+
+==Motivation==
+
+Quantum computing poses a fundamental threat to elliptic curve cryptography.  
+Shor’s algorithm renders discrete logarithm-based signatures (ECDSA, Schnorr) insecure.  
+
+NIST has standardized post-quantum signatures such as CRYSTALS-Dilithium, but direct integration into Bitcoin faces obstacles:
+* Large signature sizes (2–5 KB vs. 64 bytes for Schnorr).
+* Different verification algorithms requiring new opcodes.
+* Compatibility issues with Taproot and existing Bitcoin consensus.
+
+'''PQ-Derived Schnorr''' addresses these by:
+* Using Dilithium off-chain for quantum resistance.
+* Using Schnorr on-chain for efficiency and compatibility.
+* Binding both signature components cryptographically to prevent mix-and-match attacks.
+
+==Specification==
+
+===1. Key Generation===
+* Generate Dilithium key pair: <code>(pk_dil, sk_dil)</code>.
+* Generate Schnorr key pair: <code>(pk_sch, sk_sch)</code>.
+* Hybrid key: <code>pk = (pk_dil, pk_sch)</code>.
+
+===2. Signing===
+# Compute Dilithium signature <code>(c, z)</code> on message <code>m</code>.
+# Compute binding hash <code>h = H2(c || m)</code>.
+# Generate Schnorr signature <code>(R, s)</code> on <code>h</code>.
+# Hybrid signature: <code>σ = ((c, z), (R, s))</code>.
+
+===3. Verification===
+; Classical Mode (default)
+: Compute <code>h = H2(c || m)</code>.  
+: Verify <code>(R, s)</code> with <code>pk_sch</code>.  
+: Ensures lightweight on-chain validation (BIP340-compatible).  
+
+; Full Hybrid Mode (Tier-2)
+: Verify Dilithium <code>(c, z)</code> with <code>pk_dil</code>.  
+: Verify Schnorr <code>(R, s)</code> with <code>pk_sch</code>.  
+: Provides full post-quantum security.
+
+===4. Bitcoin Script Integration===
+* '''Taproot Leaf Extension'''  
+<syntaxhighlight lang="text">
+OP_1 <pubkey_combined> OP_CHECKSIG_HYBRID
+</syntaxhighlight>
+
+: <code>pubkey_combined = (pk_dil, pk_sch)</code>  
+: <code>OP_CHECKSIG_HYBRID</code>: verifies Schnorr on-chain, binds Dilithium off-chain.
+
+* '''Soft Fork Opcode'''  
+Introduce <code>OP_CHECKSIG_QES2</code>:  
+<syntaxhighlight lang="text">
+Stack: [signature, pubkey]
+Parse σ = ((c,z),(R,s)), pk = (pk_dil, pk_sch)
+h = H2(c||message)
+result = Schnorr.Verify(pk_sch, h, (R,s))
+Push result
+</syntaxhighlight>
```

Comment:
Does not render as intended:

<img width="1016" height="65" alt="Image" src="https://github.com/user-attachments/assets/69d73c92-a6d5-49e7-8917-72e597508495" />